spidermon/contrib/scrapy/extensions.py

```diff
@@ -238,5 +240,6 @@ def _generate_data_for_spider(self, spider):
             else [],
             "crawler": self.crawler,
             "spider": spider,
+            "sc_spider_name": os.getenv("SHUB_VIRTUAL_SPIDER", spider.name),
```

Comment:
```suggestion
            "sc_spider_name": os.getenv("SHUB_VIRTUAL_SPIDER") or spider.name,
``` 

Otherwise `SHUB_VIRTUAL_SPIDER=""` will cause `sc_spider_name` to be `""`